### ISSUE TITLE
[help needed] How to apply a style to a separate stream?

### ISSUE THREAD

**Issue by askhat** [opentok/opentok-react-native#193]
I'm trying to build an app for multi-user conferencing. Any user may or may not have a priority over others, expressed with this user's stream being bigger. For example, a layout for 3+1 users may look like so:

```
_________
|   |___|
|___|___|
```

With left tile allocated for the prioritized stream and the smaller tiles for the rest.

In the `opentok-react` it is easily achieved with iterating over streams array (which is made with `createSession` script) and mapping each stream to `OTSubscriber`, so each subscriber may be sized separately.

However, I was not able to implement it with `opentok-react-native`. It's not clear either how to size each separate stream or how to iterate as above.

I assume there is a way, but I don't quite get how to do it. Would anyone please explain how do I manage to achieve the desired behavior?

**Comment by mbouxin** [comment]
@askhat I did not do it yet but as I have also this feature to be developed by the beginning of 2019 I already thought a little about this.

As it is now, in opentok-react-native, you should:
 - listen to streamCreated and streamDestroyed event of the OTSession and save yourself a list of active streams
 - create as many OTSubscriber as active streams with the style props particularized

Obviously you can do it "manually" or modify OTSession to register it streams and create a OTStreams as what was done in the opentok-react project

**Comment by malkaShtibel** [comment]
Can you provide a code example of creating a subscriber dynamically on streamCreated event?
I cannot understand how I am supposed to 'create as many OTSubscriber as active streams with the style props particularized'

**Comment by msach22** [comment]
@askhat @malkaShtibel Have you tried  doing something like this:
https://github.com/opentok/opentok-react-native/issues/124#issuecomment-403996366

It shows an example of a button for each subscriber, but you can use the same concept for other styles.

**Comment by msach22** [comment]
This will be addressed with v0.9.8

**Comment by msach22** [comment]
This was released with v0.9.8

**Comment by og2t** [comment]
Hi Manik,

Thanks for the example above. However, when we try to create a few instances of the `OTSubscriber` in one `OTSession` the following error occurs:

```
OTRN JS: There was an error:  Object {
 “message”: “Error subscribing. Could not find native stream for subscriber.“,
}
```

It seems it would be much easier to pass a custom render method to `OTSubscriber` to display each subscriber, rather than manage subscribers manually by creating multiple `OTSubscriber` components.

The feature is described [in this issue](https://github.com/opentok/opentok-react-native/issues/287), what do you think?
